product_hunt_utils.py

```diff
@@ -69,6 +68,173 @@ def get_commentor_info_for_product_id(product_id: int)-> list:
+
+    first_response = first_response.json()
+
+    first_followers = first_response["data"]["profile"]["followers"]["edges"]
+    end_cursor = first_response["data"]["profile"]["followers"]["pageInfo"]["endCursor"]
+    next_page = first_response["data"]["profile"]["followers"]["pageInfo"]["hasNextPage"]
+
+    for nodes in first_followers:
+          id_lst.append(nodes["node"]["id"])
+
+    while(next_page == True):
+              json_data = {
+                'operationName': 'ProfileFollowersQuery',
+                'variables': {
+                    'username': person_username,
+                    'cursor': end_cursor,
+                    'query': '',
+                },
+                'query': 'query ProfileFollowersQuery($username:String!$cursor:String){profile:user(username:$username){id followersCount followers(first:20 after:$cursor){edges{node{id ...UserItemFragment __typename}__typename}pageInfo{endCursor hasNextPage __typename}__typename}...MetaTags __typename}}fragment MetaTags on SEOInterface{id meta{canonicalUrl creator description image mobileAppUrl oembedUrl robots title type author authorUrl __typename}__typename}fragment UserItemFragment on User{id name headline username followersCount ...UserImage ...UserFollowButtonFragment __typename}fragment UserFollowButtonFragment on User{id followersCount isFollowed __typename}fragment UserImage on User{id name username avatarUrl __typename}',
+            }
+
+              response = requests.post('https://www.producthunt.com/frontend/graphql', cookies=cookies, headers=headers, json=json_data)
+
+              response = response.json()
+
+              followers = response["data"]["profile"]["followers"]["edges"]
+
+              end_cursor = response["data"]["profile"]["followers"]["pageInfo"]["endCursor"]
+
+              next_page = response["data"]["profile"]["followers"]["pageInfo"]["hasNextPage"]
+
+              for nodes in followers:
+                    id_lst.append(nodes["node"]["id"])
+
+    df = pd.DataFrame(id_lst, columns = ["person id"])
+
+
+    df.to_csv("person_followers_ids", index = False)
+
+    df = pd.read_csv("person_followers_ids")
+
+    follow_count = 0
+
+    while(True):
+
+        for i,r in df.iterrows():
+            time.sleep(5)
+            if(follow_a_person(int(r["person id"]))):
+                    follow_count += 1
+                    df = df.drop(i)
+            else:
+                    break
+        if (df.empty):
+              print("All followers are followed")
+              os.remove("person_followers_ids")
+              return True
+        df.to_csv("person_followers_ids",index = False)
+        print("followed count: ",follow_count)
+        print("wait before start following again: 1 Hr")
+        time.sleep(3600)
+        df = pd.read_csv("person_followers_ids")
+    return False            

```

Comment:
this part is common in both `follow_persons_followers` and `follow_persons_followings`, it is better you refactor it into a separate function and then call it in  `follow_persons_followers` and `follow_persons_followings`. 
This way `follow_persons_followers` and `follow_persons_followings` will just hit respective APIs and create a list of ids